ui/src/timeMachine/constants/queryBuilder.ts

```diff
@@ -1,77 +1,82 @@
-import {WINDOW_PERIOD, OPTION_NAME} from 'src/variables/constants'
+export const AGG_WINDOW_AUTO = 'auto'
+export const AGG_WINDOW_NONE = 'none'
 
 export interface QueryFn {
   name: string
-  flux: string
+  flux: (period?: string) => string
 }
 
 export const FUNCTIONS: QueryFn[] = [
   {
     name: 'mean',
-    flux: `|> aggregateWindow(every: ${OPTION_NAME}.${WINDOW_PERIOD}, fn: mean)`,
+    flux: period => `|> aggregateWindow(every: ${period}, fn: mean)`,
   },
   {
     name: 'median',
     // TODO: https://github.com/influxdata/influxdb/issues/13806
-    flux: `|> window(period: ${OPTION_NAME}.${WINDOW_PERIOD})
+    flux: period => `|> window(period: ${period})
```

Comment:
Once this `median` function doesn't require a special case (which I think will be within the next Flux release), I bet we can move all this templating logic into `src/timeMachine/utils/queryBuilder`.